src/darray.h

```diff
@@ -143,6 +143,13 @@ typedef darray (unsigned long)  darray_ulong;
     (arr).item[--(arr).size] = 0; \
 } while (0)
 
+#define darray_remove(arr, i) do { \
+	size_t index_ = (i); \
+	if (index_ < arr.size-1)    \
```

Comment:
style nitpick:
```suggestion
	if (index_ < arr.size-1) \
```

also, `arr.size` and `size_t` are both unsigned so removing any index from a zero-sized array will proceed without a warning (or rather: it's not a noop as expected). Might be better to use `ssize_t` here to check for `<0` as well.

I strongly recommend splitting the `darray_remove` addition into a separate patch and adding test cases for it so we don't have to worry about this. There's nothing existing that I can find, but a `test/darray.c` won't hurt :)

Reply:
Done